### PR TITLE
Marketplace Thank You: Update plugin card styles

### LINKED ISSUE #74740: Marketplace Thank You: Update Plugins card style
Figma design 5ddl1WFTEnaIatL2WlZp4k-fi-2596%3A11300

### PR DESCRIPTION
**Pull request by epeicher**
Resolves #74740

**Depends on:**
- https://github.com/Automattic/wp-calypso/issues/74739

Once the following PR is merged. There should not be a border-top in the first plugin card:
- https://github.com/Automattic/wp-calypso/issues/74739

That will remove the extra section created when there are no themes

## Proposed Changes

* Update styles for Plugin cards in Marketplace Thank You page

## Testing Instructions

* Go to `/plugins:siteId` and purchase a couple of Marketplace plugins
* Make sure the design of the Thank You page plugin cards matches 5ddl1WFTEnaIatL2WlZp4k-fi-2596%3A11300

|Before | After|
|-------|------|
| CleanShot 2023-03-22 at 18 02 35@2x | CleanShot 2023-03-22 at 18 07 28@2x    |

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Comment by epeicher** [comment]
Hi @vinimotaa , I have implemented the plugin card styles. The scenario with only plugins currently has a `border-top` for all the cards. Removing that border would require adding some extra logic or complex CSS selectors. Do you think this is good for the first iteration?

|Plugins and a Theme | Only plugins|Only a Theme|
|-------|------|------|
|   CleanShot 2023-03-23 at 12 39 04@2x     |      CleanShot 2023-03-23 at 12 38 40@2x  |    CleanShot 2023-03-23 at 12 39 23@2x    |

UPDATE: The above will be fixed as part of [the ](https://github.com/Automattic/wp-calypso/pull/74792/commits/f1ac0fc58f3f1628dfdf4212b7045dcae42a79b4)

File: client/my-sites/checkout/checkout-thank-you/marketplace/marketplace-thank-you-plugin-section.tsx [hunk 1]
```diff
@@ -22,11 +23,10 @@ const PluginSectionContainer = styled.div`
 	flex-direction: row;
 	flex-wrap: wrap;
 	width: 720px;
-	padding: 24px;
 	box-sizing: border-box;
-	border: 1px solid var( --studio-gray-5 );
-	border-radius: 4px;
+	border-top: 1px solid var( --studio-gray-5 );
```
Comment (cpapazoglou) [inline client/my-sites/checkout/checkout-thank-you/marketplace/marketplace-thank-you-plugin-section.tsx]: It seems that we need to move the styling to https://github.com/Automattic/wp-calypso/blob/732eadac995183afa5acbabab5dbed64037ad72d/client/components/thank-you/index.tsx#L29-L31

CleanShot 2023-03-23 at 14 05 01@2x

You can try it by visiting `/checkout/gift/thank-you/:site`

Comment (cpapazoglou) [reply client/my-sites/checkout/checkout-thank-you/marketplace/marketplace-thank-you-plugin-section.tsx]: Doing this accompanied with a `:first-child` rule should be enough for getting rid of the line of the first element.

**Review by cpapazoglou — CHANGES_REQUESTED** [review]
To be honest, the top border of the first item looked like a glitch to me before reading the comments. I believe fixing it doesn't require enormous effort.

File: client/my-sites/checkout/checkout-thank-you/marketplace/marketplace-thank-you-plugin-section.tsx (hunk 1 above)
Comment (epeicher) [reply client/my-sites/checkout/checkout-thank-you/marketplace/marketplace-thank-you-plugin-section.tsx]: This should be sorted by [f1ac0fc](https://github.com/Automattic/wp-calypso/pull/74792/commits/f1ac0fc58f3f1628dfdf4212b7045dcae42a79b4) and the merge that includes this commit https://github.com/Automattic/wp-calypso/commit/e05ecee17cefeba722636cd9ef73899d83b056a7 that removes the theme section when there are no themes to display [here](https://github.com/Automattic/wp-calypso/commit/e05ecee17cefeba722636cd9ef73899d83b056a7#diff-ffe5b17b99241dfd054caea592cb72136e81dce734f9f62f5ae1b0ff3843b13dR121)

**Review by cpapazoglou — APPROVED** [review]
Thanks @epeicher, looks stunning now

CleanShot 2023-03-27 at 17 44 19@2x


CleanShot 2023-03-27 at 17 40 31@2x

**Pull request merged**